core/src/main/java/nl/nn/adapterframework/http/rest/ApiDispatchConfig.java

```diff
@@ -15,21 +15,22 @@
 */
 package nl.nn.adapterframework.http.rest;
 
-import java.util.HashMap;
+import java.util.Collections;
 import java.util.Map;
 import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
 
 import nl.nn.adapterframework.core.ListenerException;
 
 public class ApiDispatchConfig {
 	private String uriPattern;
-	private Map<String, ApiListener> methods = new HashMap<String, ApiListener>();
+	private Map<String, ApiListener> methods = new ConcurrentHashMap<String, ApiListener>();
 
 	public ApiDispatchConfig(String uriPattern) {
 		this.uriPattern = uriPattern;
 	}
 
-	public void register(String method, ApiListener listener) throws ListenerException {
+	public synchronized void register(String method, ApiListener listener) throws ListenerException {
```

Comment:
Hier zou je in plaats van deze synchronized een computeIfAbsent kunnen doen, en dan daarna checken of de registreerde listener overeenkomt met de paramter listener

Reply:
Maar dan wordt het gooien van de exceptie lelijk :(